zebra-consensus/src/block/subsidy/funding_streams.rs

```diff
@@ -0,0 +1,47 @@
+//! Funding Streams calculations. - [§7.7][7.7]
+//!
+//! [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+
+use zebra_chain::{
+    amount::{Amount, Error, NonNegative},
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+};
+
+use crate::{
+    block::subsidy::general::block_subsidy,
+    parameters::subsidy::{
+        FundingStreamReceiver, FUNDING_STREAM_HEIGHT_RANGES, FUNDING_STREAM_RECEIVER_DENOMINATOR,
+        FUNDING_STREAM_RECEIVER_NUMERATORS,
+    },
+};
+
+#[cfg(test)]
+mod tests;
+
+/// Returns the `fs.Value(height)` for each stream receiver
+/// as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+use std::collections::HashMap;
+pub fn funding_stream_values(
+    height: Height,
+    network: Network,
+) -> Result<HashMap<FundingStreamReceiver, Amount<NonNegative>>, Error> {
+    let canopy_height = Canopy.activation_height(network).unwrap();
+    let mut results = HashMap::new();
+
+    if height >= canopy_height {
+        let range = FUNDING_STREAM_HEIGHT_RANGES.get(&network).unwrap();
+        if range.contains(&height) {
+            let block_subsidy = block_subsidy(height, network)?;
+            for (&receiver, &numerator) in FUNDING_STREAM_RECEIVER_NUMERATORS.iter() {
+                let amount_value =
+                    ((block_subsidy * numerator)? / FUNDING_STREAM_RECEIVER_DENOMINATOR)?;
```

Comment:
https://github.com/zcash/zcash/blob/master/src/consensus/funding.cpp#L46

We need to make sure this is also true in rust.

Reply:
In Rust, "integer division rounds towards zero":
https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html#arithmetic-and-logical-binary-operators

This is the same as `floor`, because these numbers are all positive.

Reply:
Thank you for the reference. Do you think we should have a doc comment clarifying we are relying in this rust feature just as zcashd do ?

Reply:
I think it would be helpful to quote the equation from the Zcash spec, and the integer division rounding rule from the Rust spec.

Reply:
Added some notes at https://github.com/ZcashFoundation/zebra/pull/3017/commits/c37ef82f5023aed88b060b6fb4dd2a256a1ebf9e